openstack/identity/v3/extensions/endpointgroups/results.go

```diff
@@ -0,0 +1,76 @@
+package endpointgroups
+
+import (
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// Extract interprets a GetResult, CreateResult or UpdateResult as a concrete
+// EndpointGroup. An error is returned if the original call or the extraction
+// failed.
+func (r commonResult) Extract() (*EndpointGroup, error) {
+	var s struct {
+		EndpointGroup *EndpointGroup `json:"endpoint_group"`
+	}
+	err := r.ExtractInto(&s)
+	return s.EndpointGroup, err
+}
+
+// GetResult is the response from a Get operation. Call its Extract method
+// to interpret it as an EndpointGroup.
+type GetResult struct {
+	commonResult
+}
+
+// EndpointFilter represents a set of one or several criteria to match endpoints.
+type EndpointFilter struct {
```

Comment:
If the above comment is true, we can get rid of this struct. When POST/create is implemented, we can do something like [this](https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/flavors/requests.go#L22-L35) to help with the input validation.